src/main/java/org/assertj/core/api/AbstractAssert.java

```diff
@@ -72,6 +72,8 @@
 
   protected Objects objects = Objects.instance();
 
+  protected Predicate<ACTUAL> ignoreWhen = actual -> false;
```

Comment:
I would simply set it to `null` by default so that we don't have to execute a predicate for nothing, just need a `null` check